server/modules/include/schemarouter.h

```diff
@@ -290,6 +298,13 @@ struct router_client_session {
         ROUTER_STATS    stats;     /*< Statistics for this router         */
         int             n_sescmd;
         int             pos_generator;
+	/** for prepared statement handling */
+        char            *rses_last_tname; /*< prepare statement target */
+        bool            rses_sent_prepare; /*< prepare statement was sent */
+        int			        rses_next_stmtid; /*< next stmtid to use */
+        int			        rses_max_stmtid; /*< stmtid map size */
+        PREP_MAP**			rses_prep_map; /*< mapping client stmtid to server, server stmtid */
```

Comment:
Instead of using an array, you could use MaxScale's hashtable implementation found in server/core/hashtable.c. This could allow easier mapping of statement ID's to server names.